### DISCUSSION TITLE
Creating / Modifying a template for generating a quasar app.

### DISCUSSION THREAD

**Discussion by peppermintbunny** [quasarframework/quasar#15062] [Show and Tell]
Where can i find information on how to create or modify the basic template Quasar uses to create an app?  Ive seen a super template at https://github.com/MilosPaunovic and I'd like to create my own.  What tools are used to create a template and how do you use or "run" the template to create a new customised app?

**Comment by MilosPaunovic** [comment]
I might be able to help you, but I'm gonna need a bit more of an explanation of what exactly are you looking to achieve, as I don't understand it from the post above?

**Reply by peppermintbunny** [reply]
HI Milos. Thanks for reaching out!!!  I'm new to frontend development using javascript, html and css so my questions could seem elementary.  I want to streamline the app Quasar initially creates.  I always remove Essential Links component and the hard-coded links to external sites, add some of my own components and create a folder for composables. I add dialog and notify to quasar.config. So I was wanting modify how Quasar creates its initial app.  I think I need to replace the template Quasar's CLI uses to create the app with my own template (which would be very similar to your version) and thereby be able to create an inital app to my specification.  However, I don't know how to do this.  Could you point me in the right direction?  Do I find where Quasar's template has been installed on my imac?  Do I run Webpack or Vite?  Or do I do something else?

**Reply by MilosPaunovic** [reply]
There were couple of discussions in the Quasar team to enable some sort of setting defaults for starting new projects, but that is still in the idea phase. If you are a beginner, I would suggest that you create a 'base' template of how you want your starter projects to work and to save it to GH as a template project. It is the simplest and quickest solution which will give you the ability to start new projects in a matter of seconds.

**Reply by peppermintbunny** [reply]
And once I have my 'base' template saved in GH, how do I create a template project using my base template? I realise this may be a big ask, so if it is easier, please direct me to some documentation.  Many thanks for your help.

**Reply by MilosPaunovic** [reply]
One you have it saved as a template, you're gonna have `Use this template` button for creating new repositories from that template.
![image](https://user-images.githubusercontent.com/62475782/206425438-c1766792-b03b-4027-ab2a-977724f3ecf4.png)

**Reply by peppermintbunny** [reply]
Oh what magic!  Ive just tried it using your template and it worked.  Thank you very much for your kind help.
